File: solutions/typescript/eliuds-eggs/2/eliuds-eggs.ts
//
// This is only a SKELETON file for the 'Pop Count' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const eggCount = (displayValue: number): number => {
  let ans = 0
  let n = displayValue
  while (n != 0) {
    if (n % 2 = 1) {
      ans += 1
    }
    n /= 2
  }
  return ans
}

File: solutions/csharp/eliuds-eggs/1/EliudsEggs.cs
public static class EliudsEggs
{
    public static int EggCount(int encodedCount)
    {
        var ans = 0;
        var e = encodedCount;
        while (e != 0)
        {
            ans += e & 1;
            e >>= 1;
        }
        return ans;
    }
}
